dateparser/conf.py

```diff
@@ -31,19 +31,21 @@ class Settings:
     _pyfile_data = None
     _mod_settings = dict()
 
-    def __init__(self, settings=None):
-        if settings:
-            self._updateall(settings.items())
-        else:
+    def __init__(self, **kwargs):
+        if not kwargs.get('settings'):
             self._updateall(self._get_settings_from_pyfile().items())
 
+        elif len(self.__dict__) == 1:
```

Comment:
I find this hard to read, as I have no idea what `self.__dict__` is meant to contain at this point. What is this checking for? Any way we can make this check more readable?

Reply:
On each init Settings checks incoming params and create new instance or take one from `__registry_dict`.\
New ones on init have only one attribute: `registry_key` and others are set by `self._updateall`.\
This check is checking if it's a new instance to avoid updating existing existing instances as this does not make any good and leads to concurrency issues.\
Not sure we can make this check any nicer..